src/classes/ALLO_Allocations_TDTM.cls

```diff
@@ -536,149 +536,161 @@ public class ALLO_Allocations_TDTM extends TDTM_Runnable {
     private void copyRecurringDonationCampaignAndPaymentAllocations(List<SObject> listSObjs) {
 
         for (SObject sObj : listSObjs) {
+            Boolean isPaymentRecord = sObj instanceof npe01__OppPayment__c;
 
-            //if we have recurring donation or campaign allocations
-            if ((sObj instanceof npe01__OppPayment__c && mapWrapper.containsKey((Id)sObj.get('npe01__Opportunity__c'))) || !(sObj instanceof npe01__OppPayment__c) && ((sObj.get('npe03__Recurring_Donation__c') != null && mapWrapper.containsKey((Id)sObj.get('npe03__Recurring_Donation__c'))) ||
-                (sObj.get('CampaignId') != null && mapWrapper.containsKey((Id)sObj.get('CampaignId'))))) {
+            if (isPaymentRecord && mapWrapper.containsKey((Id)sObj.get('npe01__Opportunity__c'))) {
+                copyRecurringDonationCampaignAndPaymentAllocations(sObj);
 
-                alloWrapper wrap;
-                Map<Id, Allocation__c> GAUtoAlloMapping = new Map<Id, Allocation__c>();
-                Boolean isPaymentAllocation = false;
+            } else if (sObj.get('npe03__Recurring_Donation__c') != null && mapWrapper.containsKey((Id)sObj.get('npe03__Recurring_Donation__c'))) {
```

Comment:
I think you need to keep the `!isPaymentRecord &&` here and the next if. For example, if this is a Payment Record (and it did not get caught in the first if), then this line will throw an exception because there is no `npe03__Recurring_Donation__c` field no the Payment object.

Reply:
Not sure about that.  mapWrapper is Id to alloWrapper and alloWrapper is this...

```
        boolean isPercentOnly = true;                                        //tracks whether all allocations are percentage based
        Decimal parentAmount = null;                                         //amount of this allocation's parent
        Decimal totalAmount = 0;                                             //holds total allocation amount
        Decimal totalPercent = 0;                                            //holds total percentage allocations
        List<Allocation__c> triggerList = new List<Allocation__c>();         //all allocations for this opportunity in the current trigger set
        List<Allocation__c> listAllo = new List<Allocation__c>();            //all non-default allocations for this parent object
        Map<Id, Allocation__c> defaultDupesById = new Map<Id, Allocation__c>();//duplicate default allocations, if any
        Allocation__c defaultAllo = null;                                    //the default allocation for this parent object
        boolean defaultInTrigger = false;                                    //is the default allocation in the current trigger set?

```

Don't see how that ever throw an exception.  fillWrapper contains the same fields for any object, so if we have sobject, we pretty much have to have those fields don't we?

Reply:
sorry, you're right.  will fix.